unit-tests/py/rspy/usb_relay.py

```diff
@@ -0,0 +1,268 @@
+        :param ports: List of port numbers; if not provided, enable all ports
+        :param disable_other_ports: if True, the ports not in the list will be disabled
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.enable_ports(ports, disable_other_ports, sleep_on_change)
+        return False
+
+    def disable_ports(self, ports=None, sleep_on_change=0):
+        """
+        :param ports: List of port numbers; if not provided, disable all ports
+        :param sleep_on_change: Number of seconds to sleep if any change is made
+        :return: True if no errors found, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.disable_ports(ports, sleep_on_change)
+        return False
+
+    def recycle_ports(self, portlist=None, timeout=2):
+        """
+        Disable and enable a port
+        :param portlist: List of port numbers; if not provided, recycle all ports
+        :param timeout: how long to wait before re-enabling
+        :return: True if everything OK, False otherwise
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.recycle_ports(portlist=portlist, timeout=timeout)
+        return False
+
+    def find_all_hubs(self):
+        """
+        Yields all hub port numbers
+        """
+        if self._relay_type != NO_RELAY:
+            return self._relay.find_all_hubs()
+
+    @property
+    def has_relay(self):
+        return self._relay_type != NO_RELAY
+
+    @property
+    def not_supports_port_mapping(self):
+        """
+        :return: whether the connected relay supports port mapping, currently only the acroname supports it
+        """
+        return self._relay_type != ACRONAME
+
+    def get_usb_and_port_location(self, physical_port, hubs):
+        usb_location = None
+        port = None
+        try:
+            usb_location = _get_usb_location( physical_port )
+        except Exception as e:
+            log.e( 'Failed to get usb location:', e )
+
+        if self.has_relay:
+            try:
+                port = _get_port_by_loc( usb_location, hubs, self._relay, self._relay_type )
+            except Exception as e:
+                log.e( 'Failed to get device port:', e )
+                log.d( '    physical port is', physical_port )
+                log.d( '    USB location is', usb_location )
+
+        return usb_location, port
+
+
+#################################
+def _find_active_relay():
+    """
+    Function finds an available relay to connect to and returns it
+    """
+    acroname_relay = _is_there_acroname()
+    if acroname_relay:
+        return ACRONAME, acroname_relay
+
+    ykush_relay = _is_there_ykush()
+    if ykush_relay:
+        return YKUSH, ykush_relay
+
+    return NO_RELAY, None
```

Comment:
The previous code printed out some debug info if nothing was found (sys.path, I think). Let's keep the functionality.